Source/WebCore/Modules/cookie-store/CookieStore.cpp

```diff
@@ -192,14 +192,36 @@ void CookieStore::set(CookieInit&& options, Ref<DeferredPromise>&& promise)
     cookieJar.setCookieAsync(document, url, cookie, WTFMove(completionHandler));
 }
 
-void CookieStore::remove(const String&, Ref<DeferredPromise>&& promise)
+void CookieStore::remove(String&& name, Ref<DeferredPromise>&& promise)
 {
-    promise->reject(NotSupportedError);
+    remove(CookieStoreDeleteOptions { WTFMove(name), { } }, WTFMove(promise));
 }
 
-void CookieStore::remove(CookieStoreDeleteOptions&&, Ref<DeferredPromise>&& promise)
+void CookieStore::remove(CookieStoreDeleteOptions&& options, Ref<DeferredPromise>&& promise)
 {
-    promise->reject(NotSupportedError);
+    auto* context = scriptExecutionContext();
+    if (!context) {
+        promise->reject(SecurityError);
+        return;
+    }
+
+    auto* origin = context->securityOrigin();
+    if (!origin) {
+        promise->reject(SecurityError);
+        return;
+    }
+
+    if (origin->isOpaque()) {
+        promise->reject(Exception { SecurityError, "The origin is opaque"_s });
+        return;
+    }
+
+    CookieInit initOptions;
+    initOptions.name = WTFMove(options.name);
+    initOptions.value = ""_s;
+    initOptions.expires = 0;
```

Comment:
Doesn't this end up creating a negative value in the dictionary we send to CFNetwork. I think we should use `WallTime::now() - 24h`

Reply:
I think you're right. It may become negative when converted to a NSHTTPCookie. I can try `WallTime::now() - 24h`. The spec says that although the time doesn't matter as long as it's in the past, ideally we'd like to set it to the earliest representable date (https://wicg.github.io/cookie-store/#delete-a-cookie). Is there a good way to do this or do we prefer to just use `WallTime::now() - 24h`?

Reply:
I stand by recommendation because we don't seem to have a good way to get the earliest representable date. Also, it doesn't matter as long as it is in the past.

Reply:
Sounds good!